### PR TITLE
Fix: Persist LabelEncoder along with XGBoost model

### LINKED ISSUE #14: XGBoost model returns numeric codes instead of crop names due to missing LabelEncoder persistence
Description
The XGBoost model currently uses LabelEncoder to convert crop names (string labels) into integers during training.
However, only the trained model is saved to disk — the LabelEncoder itself is not persisted.
As a result, when the model is reloaded for inference, it outputs encoded integers (e.g., 0–21) instead of human-readable crop names.
Impact
Severity: High
Breaks production deployment
Makes the trained XGBoost model unusable without retraining
Affects API endpoints depending on decoded crop names
Farmers receive numeric codes instead of actual crop names
Expected Behavior
Both the trained model and the LabelEncoder should be saved and reloaded together so predictions can be decoded back to original crop names.
Proposed Solution
Save the model and encoder together in one file (or separately if preferred).
Example implementation:
import joblib
# During training
model.fit(X_train, y_train)
joblib.dump({'model': model, 'encoder': label_encoder}, 'crop_model.pkl')
# During inference
data = joblib.load('crop_model.pkl')
model = data['model']
encoder = data['encoder']
decoded_preds = encoder.inverse_transform(model.predict(X_test))
Additional Context
This issue impacts model deployment and user-facing outputs.
Please consider adding the hacktoberfest label.
I’d also like to work on fixing this issue myself. 🙌

### PR DESCRIPTION
**Pull request by shupandee**
- Saves both the trained model and LabelEncoder using joblib
- Ensures predictions are decoded back to crop names during inference
- Prevents numeric outputs in production

Closes #14
Screenshot 2025-10-23 230944
Screenshot 2025-10-23 230928
<issue_number>

### CODE REVIEW

**Comment by shupandee** [comment]
@vannu07 
Thank you so much! I really appreciate the feedback.
I’m glad the fix helps improve the model’s usability and decoding of crop names.
Looking forward to contributing more in the future!
